libraries/chain/chain_controller.cpp

```diff
@@ -717,6 +718,16 @@ void chain_controller::update_global_properties(const signed_block& b) {
          gpo.active_producers = std::move(schedule);
          gpo.configuration = std::move(config);
       });
+
+      auto active_producers_authority = types::Authority(uint32_t(gpo.active_producers.size()*config::ProducersAuthorityThreshold), {}, {});
```

Comment:
This like will break once you replace the double constant with a constant number of producers threshold.

The active producers size should always be 21.